### PR TITLE
Common - Use CBA addons instead of activatedAddons

### PR DESCRIPTION
**Pull request by dedmen**
Ref: https://github.com/CBATeam/CBA_A3/pull/1051#issuecomment-461881408

Related: #6800

### CODE REVIEW

**Comment by commy2** [comment]
Does this break anything due to case sensitivity?

**Comment by dedmen** [comment]
We look for `a3_` I think it's `A3_`

**Comment by dedmen** [comment]
Yes. `cba_common_addons` -> `A3_Functions_F`

Also find a3_ is kinda bad. Shouldn't we just select the first 3 characters and compare if it's a3_ or ace?

**Comment by dedmen** [comment]
How about this?

File: addons/common/scripts/checkVersionNumber.sqf [hunk 1]
```diff
@@ -4,10 +4,11 @@
 private _files = [];
 
 {
-    if (_x find "a3_" != 0 && {_x find "ace_" != 0} && {!(toLower _x in (missionNamespace getVariable ["ACE_Version_Whitelist", []]))}) then {
+    private _start = toLower (_x select [0,3]);
+    if (!(_x in ["a3_", "ace"]) && {!(toLower _x in (missionNamespace getVariable ["ACE_Version_Whitelist", []]))}) then {
```
Comment (PabstMirror) [inline addons/common/scripts/checkVersionNumber.sqf]: `acex_thing` will now be filtered

Comment (PabstMirror) [reply addons/common/scripts/checkVersionNumber.sqf]: also `_start` isn't used?

Comment (dedmen) [reply addons/common/scripts/checkVersionNumber.sqf]: :man_facepalming:

File: addons/common/scripts/checkVersionNumber.sqf [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
-private _files = [];
-
-{
-    if (_x find "a3_" != 0 && {_x find "ace_" != 0} && {!(toLower _x in (missionNamespace getVariable ["ACE_Version_Whitelist", []]))}) then {
-        _files pushBack _x;
-    };
-} forEach activatedAddons;
+private _aceWhitelist = missionNamespace getVariable ["ACE_Version_Whitelist", []];
+private _files = CBA_common_addons select {
+    private _x = toLower _x;
```
Comment (commy2) [inline addons/common/scripts/checkVersionNumber.sqf]: _x is already private in this scope.

File: addons/common/scripts/checkVersionNumber.sqf [hunk 3]
```diff
@@ -1,13 +1,13 @@
...
+    ((_x select [0,3]) != "a3_") &&
+    {(_x select [0,4]) != "ace_"} &&
```
Comment (commy2) [inline addons/common/scripts/checkVersionNumber.sqf]: too many parenthesii

Comment (dedmen) [reply addons/common/scripts/checkVersionNumber.sqf]: Make suggestions and I'll click on the commit suggestion button.

Comment (jonpas) [reply addons/common/scripts/checkVersionNumber.sqf]:
```suggestion
    {_x select [0,4] != "ace_"} && 
```
For @commy2.

File: addons/common/scripts/checkVersionNumber.sqf [hunk 4]
```diff
@@ -1,13 +1,12 @@
...
-private _files = [];
-
-{
-    if (_x find "a3_" != 0 && {_x find "ace_" != 0} && {!(toLower _x in (missionNamespace getVariable ["ACE_Version_Whitelist", []]))}) then {
-        _files pushBack _x;
-    };
-} forEach activatedAddons;
+private _aceWhitelist = missionNamespace getVariable ["ACE_Version_Whitelist", []];
+private _files = CBA_common_addons select {
+    ((_x select [0,3]) != "a3_") &&
```
Comment (jonpas) [inline addons/common/scripts/checkVersionNumber.sqf]:
```suggestion
    (_x select [0,3] != "a3_") &&
```
For @commy2.

**Comment by dedmen** [comment]
Thanks for making commy happy

File: addons/common/scripts/checkVersionNumber.sqf [hunk 5]
```diff
@@ -1,13 +1,12 @@
...
-private _files = [];
-
-{
-    if (_x find "a3_" != 0 && {_x find "ace_" != 0} && {!(toLower _x in (missionNamespace getVariable ["ACE_Version_Whitelist", []]))}) then {
-        _files pushBack _x;
-    };
-} forEach activatedAddons;
+private _aceWhitelist = missionNamespace getVariable ["ACE_Version_Whitelist", []];
+private _files = CBA_common_addons select {
+    (_x select [0,3] != "a3_") &&
```
Comment (commy2) [inline addons/common/scripts/checkVersionNumber.sqf]:
```suggestion
    _x select [0,3] != "a3_" &&
```

File: addons/common/scripts/checkVersionNumber.sqf [hunk 6]
```diff
@@ -1,13 +1,12 @@
...
+    {_x select [0,4] != "ace_"} && 
+    {!((toLower _x) in _aceWhitelist)}
```
Comment (commy2) [inline addons/common/scripts/checkVersionNumber.sqf]:
```suggestion
   {!(toLower _x in _aceWhitelist)}
```

**Comment by commy2** [comment]
Not so fast.

File: addons/common/scripts/checkVersionNumber.sqf (hunk 6 above)
Comment (jonpas) [reply addons/common/scripts/checkVersionNumber.sqf]: I disagree, less readable.

**Comment by PabstMirror** [comment]
I accidentally merged it before we could apply those changes

**Comment by commy2** [comment]
boo

**Pull request merged**